torch/_inductor/codegen/cpp.py

```diff
@@ -687,11 +687,11 @@ def floordiv(a, b):
 
     @staticmethod
     def ceil(x):
-        return f"std::ceil({x})"
+        return f"static_cast<int64_t>(std::ceil({x}))"
 
     @staticmethod
     def trunc(x):
-        return f"std::trunc({x})"
+        return f"static_cast<int64_t>(std::trunc({x}))"
```

Comment:
Because of the changes to the lowerings, I now have to change things here.

Otherwise, `test_grid_sampler_2d_dynamic_shapes_cpu` fails while trying to use float for array subscript (in indexing):

```
/tmp/torchinductor_nkaretnikov/tq/ctqwvimpzu7iitzxzudawtqkoibfyysjpdgcoqblsky27nmadlm6.cpp:220:45: error: invalid types 'const float*[float]' for array subscript
  220 |                         auto tmp18 = in_ptr1[tmp17 + (ks1*tmp11) + (tmp1*(ks1*ks1)) + (ks2*tmp0*(ks1*ks1))];
      |                                             ^
```

Reply:
The casts belong wherever that indirect indexing expression is coming from. You can't just change the return types of ATen functions.

Reply:
@nkaretnikov This would break the vectorized kernel as well. May I know why you want to force the dtype to int64?